<!DOCTYPE html>
<html lang="en-us">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
    <title>IndexError: list index out of range (Solved) | Typophile Archive</title>
    <meta name="viewport" content="width=device-width,minimum-scale=1">
    <meta name="description" content="">
    <meta name="generator" content="Hugo 0.91.2" />
    
    
    
    
      <meta name="robots" content="noindex, nofollow">
    

    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
    <script src="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/js/bootstrap.bundle.min.js" integrity="sha384-ka7Sk0Gln4gmtz2MlQnikT1wXgYsOg+OMhuP+IlRH9sENBO0LRn5q+8nbTov4+1p" crossorigin="anonymous"></script>
    <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.1.3/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-1BmE4kWBq78iYhFldvKuhfTAU6auU8tT94WrHftjDbrCEXSU1oBoqyl2QvZ6jIW3" crossorigin="anonymous">
    <link href="/style.css" rel="stylesheet">

    <meta property="og:title" content="IndexError: list index out of range (Solved)" />
<meta property="og:description" content="" />
<meta property="og:type" content="article" />
<meta property="og:url" content="http://typophile.github.io/node/62758/" /><meta property="article:section" content="node" />
<meta property="article:published_time" content="2009-10-06T13:30:37+00:00" />
<meta property="article:modified_time" content="2009-10-06T13:30:37+00:00" />

<meta itemprop="name" content="IndexError: list index out of range (Solved)">
<meta itemprop="description" content=""><meta itemprop="datePublished" content="2009-10-06T13:30:37+00:00" />
<meta itemprop="dateModified" content="2009-10-06T13:30:37+00:00" />
<meta itemprop="wordCount" content="0">
<meta itemprop="keywords" content="" /><meta name="twitter:card" content="summary"/>
<meta name="twitter:title" content="IndexError: list index out of range (Solved)"/>
<meta name="twitter:description" content=""/>

  </head>

  <body class="ma0">

      <header>
  <nav class="navbar navbar-expand-lg navbar-light bg-light">
    <div class="container">
      <a class="navbar-brand" href="/">Typophile Archive</a>
      <button
        class="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>
    </div>
  </nav>
</header>
 
    <main role="main">
      
<article class="container">
  <header class="post-header py-3">
    <h1 class="post-title p-name" itemprop="name headline">IndexError: list index out of range (Solved)</h1>
    <p class="text-muted">
              
      <time class="f6 mv4 dib tracked" datetime="2009-10-06T13:30:37Z">October 6, 2009</time>
      
        
          • <span itemprop="author" itemscope itemtype="http://schema.org/Person">
            <span class="p-author h-card" itemprop="name">
            <a href="/?q=author:Bruno&#43;Bruno">
            Bruno Bruno
	    </a>
        </span></span>
      

    </p>
  </header>
  <div class="post-content e-content" itemprop="articleBody"><p>Hi guys,</p>
<p>Just keep getting error messages after running the following code:</p>
<code>
from robofab.world import CurrentFont
<p>myFont = CurrentFont()</p>
<p>for myGlyph in myFont:
print myGlyph.name
myGlyph.autoContourOrder()
myGlyph.correctDirection()
myGlyph.update()
for myContours in myGlyph.contours:
myContours.autoStartSegment()</p>
<p>print &ldquo;done now&rdquo;
</code></p>
<p>This is the error message:
<code>
Traceback (most recent call last):
File &ldquo;<string>&rdquo;, line 16, in ?
File &ldquo;/System/Library/Frameworks/Python.framework/Versions/2.3/lib/python2.3/site-packages/RoboFab/robofab/objects/objectsBase.py&rdquo;, line 2129, in autoStartSegment
on = segment.onCurve
File &ldquo;/System/Library/Frameworks/Python.framework/Versions/2.3/lib/python2.3/site-packages/RoboFab/robofab/objects/objectsBase.py&rdquo;, line 2330, in _get_onCurve
return self.points[-1]
File &ldquo;/System/Library/Frameworks/Python.framework/Versions/2.3/lib/python2.3/site-packages/RoboFab/robofab/objects/objectsFL.py&rdquo;, line 1942, in _get_points
node = self._node
File &ldquo;/System/Library/Frameworks/Python.framework/Versions/2.3/lib/python2.3/site-packages/RoboFab/robofab/objects/objectsFL.py&rdquo;, line 1880, in _get__node
return glyph.nodes[self._nodeIndex]
IndexError: list index out of range
</code></p>
<p>The weird thing is (after running it a second time) it will not bump into the same problem again, but rather continues to the next problem, if that makes sense. After a while the script runs fine. Really have no clue here.</p>
<p>Thanks,</p>
<p>Bruno</p>
</div>

  
   <div class="comment card m-3">
	<div class="card-body">
		<div class="commentbody"><p>Ok, for the interested: 
Following should work, cheers</p>
<code>
from robofab.world import CurrentFont
<p>myFont = CurrentFont()
print myFont.info.postscriptFullName</p>
<p>myGlyph = myFont.keys</p>
<p>for myGlyph in myFont:</p>
<blockquote>
<p>print myGlyph.name
myGlyph.autoContourOrder()
myGlyph.correctDirection()</p>
<blockquote>
<p>for myContours in myGlyph:
myContours.autoStartSegment()
myGlyph.update()</p>
</blockquote>
</blockquote>
<p>print &ldquo;done now&rdquo;
</code></p>
</div>
	</div>
	<div class="card-footer text-muted post-meta text-end">
		<time
			class="dt-published"
			datetime="2009-10-07 02:07:51"
			itemprop="datePublished"
		>
			2009-10-07 02:07:51
		</time>
		•
		<span itemprop="author" itemscope itemtype="http://schema.org/Person">

			<span class="p-author h-card" itemprop="name">
            <a href="/?q=author:Bruno&#43;Bruno">
            Bruno Bruno
	    </a>
			</span></span>
	</div>
	
</div>
 
</article>

    </main>
    
  </body>
</html>
